base/env.jl

```diff
@@ -98,6 +98,38 @@ Base.EnvDict with "N" entries:
 """
 const ENV = EnvDict()
 
+"""
+    Base.get_bool_env(name::String, default::Bool = false)::Bool
+
+Evaluate whether the value of environnment variable `name` is truthy or falsy,
+and error if none are recognized. If the variable is not set, or is set to "",
+return `default`.
+
+Recognized values are:
+    truthy: "t", "true", "y", "yes", "1"
+    falsy:  "f", "false", "n", "no", "0"
+"""
+function get_bool_env(name::String, default::Bool = false)
+    haskey(ENV, name) || return default
+    val = lowercase(get(ENV, name, ""))
+    truthy = ("t", "true", "y", "yes", "1")
+    falsy = ("f", "false", "n", "no", "0")
+    if isempty(val)
+        return default
+    elseif val in truthy
+        return true
+    elseif val in falsy
+        return false
+    else
+        error("""
+        Failed to parse the value `$val` of environment variable `$name` as Boolean.
```

Comment:
Imo, a function that deals with external data like this shouldn't throw. It should instead return something to the user indicating the error.

Otherwise, you are *always*  going to have to wrap usage of this function in a try catch since you cannot predict what is in the env var, and you are unlikely to want to panic just due to a malformed env variable.

Reply:
Seems like returning `nothing` would make sense here?

Reply:
Sounds good. I just switched it over to this